Exporting z.record fails because of keySignature when using composite

We've recently run into a typescript error when exporting schemas made with z.record, given this minimal example:

`src/index.ts`
```typescript
import * as z from "myzod";

export const myTypeSchema = z.record(z.string());
```

`tsconfig.json`
```json
{
  "compilerOptions": {
    "composite": true,
    "module": "commonjs",
    "moduleResolution": "node",
    "strict": true,
    "target": "ES6"
  },
  "include": ["src/**/*"]
}
```

This is the error:
```
src/index.ts:3:14 - error TS4023: Exported variable 'myTypeSchema' has or is using name 'keySignature' from external module "/home/johnpyp/random/myzod-bug-test/node_modules/myzod/libs/index" but cannot be named.

3 export const myTypeSchema = z.record(z.string());
               ~~~~~~~~~~~~


Found 1 error.
```

If the schema isn't exported, the error goes away.

Additionally, If `composite` is set to false, the error also goes away. `composite` is specifically used for using typescript project references so you can use individual typescript packages in a monorepo-like setting easily. The error suggests that this is part of the issue - type resolution for `keySignature` when compiling for external consumption with project references isn't possible because it isn't "named"/in scope.

https://github.com/microsoft/TypeScript/issues/5711 gives some insight on the root cause, but none of the proposed solutions here seem to work (at least in the way I interpreted them).

For now we're using a workaround that seems to work, presumably because keySignature is explicitly used/brought into scope:
```typescript
import z, { Infer } from "myzod";

export const testType = z.object({
  [z.keySignature]: z.string(),
});
```

Issue https://github.com/davidmdm/myzod/issues/42 mentions `keySignature`, and the possibility that `import * as z from 'myzod'` works when `import z from 'myzod'` doesn't, but in my testing it doesn't matter for the z.record issue at least.

--- a/src/index.ts
+++ b/src/index.ts
@@ -31,7 +31,6 @@ import {
   PartialShape,
   Eval,
   ToUnion,
-  keySignature,
   StringTypes,
   OptionalType,
   BigIntOptions,
@@ -44,7 +43,6 @@ export {
   ValidationError,
   Type,
   Infer,
-  keySignature,
   AnyType,
   ObjectShape,
   // Types
@@ -64,6 +62,9 @@ export {
   UnionType,
   IntersectionType,
 } from './types';
+
+const keySignature = Symbol('keySignature');
+export { keySignature };
 
 export const string = (opts?: StringOptions) => new StringType(opts);
 export const boolean = () => new BooleanType();

--- a/src/types.ts
+++ b/src/types.ts
@@ -1,3 +1,5 @@
+import { keySignature } from './index';
+
 function clone<T>(value: T): T {
   if (typeof value !== 'object' || value === null) {
     return value;
@@ -673,7 +675,6 @@ export class DateType extends Type<Date> implements WithPredicate<Date>, Default
   }
 }
 
-export const keySignature = Symbol('keySignature');
 export type ObjectShape = { [key: string]: AnyType; [keySignature]?: AnyType };
 
 type OptionalKeys<T extends ObjectShape> = {
@@ -694,9 +695,9 @@ type Flat<T> = T extends {} ? (T extends Date ? T : { [key in keyof T]: T[key] }
 
 type InferObjectShape<T extends ObjectShape> = Flat<
   Eval<
-    InferKeySignature<T> &
-      { [key in OptionalKeys<T>]?: T[key] extends Type<infer K> ? K : any } &
-      { [key in RequiredKeys<T>]: T[key] extends Type<infer K> ? K : any }
+    InferKeySignature<T> & { [key in OptionalKeys<T>]?: T[key] extends Type<infer K> ? K : any } & {
+      [key in RequiredKeys<T>]: T[key] extends Type<infer K> ? K : any;
+    }
   >
 >;
 
@@ -1464,9 +1465,9 @@ export type UnionOptions<T extends any[]> = {
 };
 
 type UnionIntersection<U extends UnionType<any>, T extends AnyType> = U extends UnionType<infer Schemas>
-  ? UnionType<
-      { [key in keyof Schemas]: Schemas[key] extends AnyType ? IntersectionResult<Schemas[key], T> : Schemas[key] }
-    >
+  ? UnionType<{
+      [key in keyof Schemas]: Schemas[key] extends AnyType ? IntersectionResult<Schemas[key], T> : Schemas[key];
+    }>
   : never;
 
 export class UnionType<T extends AnyType[]>

